### PR TITLE
Rework actix

### PR DESCRIPTION
**Pull request by lu-zero**
This should pave the way for testing things using the actix testing harness. See #66

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -1012,7 +988,66 @@ impl WebThingServer {
...
+    }
+
+    /// Start listening for incoming connections.
+    pub fn start(
+        &mut self,
+        configure: Option<Arc<dyn Fn(&mut web::ServiceConfig) + Send + Sync + 'static>>,
+    ) -> Server {
+        let port = self.port.unwrap_or(80);
+
+        let mut hosts = vec!["localhost".to_owned(), format!("localhost:{}", port)];
```
Comment (arpad-m) [inline src/server.rs:1003]: The code that builds `hosts` can be removed entirely, `hosts` is used nowhere in the `start` function.

**Review by arpad-m — CHANGES_REQUESTED** [review]
In addition to my inline comment `hosts.clone()` is unneccessary, one can just use `hosts`.

**Comment by lu-zero** [comment]
> In addition to my inline comment `hosts.clone()` is unneccessary, one can just use `hosts`.

I did not change much the code from its initial state, we should make a run and remove spurious clones later.

**Review by tim-hellhake — APPROVED** [review]

**Comment by tim-hellhake** [comment]
Nice cleanup!

**Pull request merged**